src/utils.cpp

```
@@ -300,7 +300,9 @@ uint32_t readReg(localArgs * la, const std::string & regName)
     uint32_t rmask = stoull(tmp[2], nullptr, 16);
     if (memhub_read(memsvc, raddr, 1, data) != 0) {
       // response->set_string("error", std::string("memsvc error: ")+memsvc_get_last_error(memsvc));
-      LOGGER->log_message(LogManager::ERROR, stdsprintf("read memsvc error: %s", memsvc_get_last_error(memsvc)));
+      if (printReadErrors) {
+        LOGGER->log_message(LogManager::ERROR, stdsprintf("read memsvc error: %s", memsvc_get_last_error(memsvc)));
+      }
```

Was this shown to have any impact on the performance in the end?
If not, I would advocate for a cleaner interface that doesn't pollute the functions.

---

I support @jsturdy 's comment. While excessive logging for sure will impact the performance, in this particular case I see no reason to provide an opportunity to hide the error message in the log. In other cases if needed this can (and should) be managed by log levels

---

The goal here is *not* to impact performance (no test that I did showed that logging slows down or speeds up any action on the card) but to make the GBT phase scan or the repeated reg read log readable.

Otherwise the log is incomprehensible because it's full of 1000's of `read memsvc error:`'s because you're potentially having bad phases during a GBT scan or your testing if the communication is good (repeated reg read) and this makes it hard to understand *anything* else.

---

So yes there *is* in my view a reason to hide this; two reasons in fact.

---

Then I think in this case, what you want is @mexanick's second point, the level here probably doesn't need to be `ERROR` (because it surely isn't treated as such here), but `WARNING` would be sufficient.
Then in a call that definitely doesn't care about spamming the log with these messages, the log level is changed before, and reverted after the call.
In the end, this codebase started as @mexanick's baby, so I'm happy for him to make the call.  If it were `cmsgemos` I'd say revert the change and do it differently..

---

So if I understand the suggestion I should pass a log level object instead.  And this would then just be given to the log manager?

---

> So if I understand the suggestion I should pass a log level object instead. And this would then just be given to the log manager?

No, one of my (possibly elided) points in this comment is that the function signature (of such low-level functions) should *not* be changed.
If I understand how the logger is working, this is what type of changes I would approve:
* Low level functions (e.g., `readReg`/`writeReg`): one of:
  1) change the level of this message to `WARNING`
  1) leave the level of this message to `ERROR` (maybe my slight semantic preference, but at some point when we go through and fix the "error" condition handling, this can be revisited, re #85/#109)
  1) remove the printing of the message entirely, and throw an exception (preference in terms of API and usage, again, for revisitation as per #85/#109)
    * forcing the calling code to handle the printing of the log message, as well as the failure
    * (but generally, I would say that any changes to this low-level functionality should be first proposed and discussed in an issue, and not added as a rider to a different feature)
* Higher-level functions (where you want to turn on/off/control this message)
  1) call to `setLogLevel(ERROR)`  (or `setLogLevel(FATAL)`, if the second method is done above) before calls to the lower level function where this message is emitted
  1) call to  `setLogLevel(DEFAULT)` (maybe one has to do a `getLogLevel` before step 1 to get this value) after you no longer want to suppress the messages
* In your higher-level function, if you want to have a parameter that enables/disables this changing of the log level (analogue of your `printReadErrors`), I have less of an issue, because it doesn't touch the core implementation of the low-level functionality (which is one of my issues with the way this change was implemented)

---

Okay; I will revert this change completely; the proposed solution you've (@jsturdy) made I think we should open a separate issue for and address it there.